storage/remote/queue_manager.go

```diff
@@ -508,7 +585,9 @@ func (t *QueueManager) sendMetadataWithBackoff(ctx context.Context, metadata []p
 
 // Append queues a sample to be sent to the remote storage. Blocks until all samples are
 // enqueued on their shards or a shutdown signal is received.
-func (t *QueueManager) Append(samples []record.RefSample) bool {
+func (t *QueueManager) Append(samples []record.RefSample, segment int) bool {
+	t.updatePendingData(len(samples), segment)
```

Comment:
this should probably happen later in the function, within the loop we may drop some samples due to relabel rules